exercises/5.1/template_ws/ros2/lesson_perception/src/perception_node.cpp

```diff
@@ -0,0 +1,211 @@
+#include "rclcpp/rclcpp.hpp"
+#include <tf2/transform_datatypes.h>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_ros/transform_broadcaster.h>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <rclcpp/qos.hpp>
+
+// PCL specific includes
+#include <pcl_conversions/pcl_conversions.h>
+#include "pcl_ros/transforms.hpp"
+
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/filters/passthrough.h>
+#include <pcl/sample_consensus/method_types.h>
+#include <pcl/sample_consensus/model_types.h>
+#include <pcl/segmentation/sac_segmentation.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl/filters/crop_box.h>
+#include <pcl/filters/statistical_outlier_removal.h>
+#include <tf2/convert.h>
+#include <tf2_eigen/tf2_eigen.h>
+#include "tf2/LinearMath/Quaternion.h"
```

Comment:
These three headers shouldn't really be in quotes.  Quotes looks in the exact directory this file is in, then falls back out to the same search path brackets use.  Mixing the two, when there isn't a local file to find, is inefficient and can be confusing.